### PR TITLE
DateBox should not raise any errors when useMaskBehavior is enabled and locale digits are different to arabic digits (T867867)

### PR DESCRIPTION
**Pull request by ksercs**
1) CheckDigit function is extended for all languages
2) Localization test modules are refactored
3) RTL for dates in globalize is removed, as in Intl
4) ca-gregorian & numbers old packages are removed from helpers. `devextreme-cldr-data` package is used now. Tests with hard-cored old data are updated.
5) Test with a formatter emulating arabic dates is moved from `excelCreator.tests` to `localization.widgets.tests`
6) `utils.formatHelper.fromat.tests` are moved to `localization.globalize`

### CODE REVIEW

File: testing/helpers/l10n/cldrCalendarDataFa.js [hunk 1]
```diff
@@ -0,0 +1,543 @@
+const Globalize = require('globalize');
+
+Globalize.load({
+    'main': {
```
Comment (San4es) [inline testing/helpers/l10n/cldrCalendarDataFa.js]: Where did we get this data?

Comment (ksercs) [reply testing/helpers/l10n/cldrCalendarDataFa.js]: From "cldr-dates-full" package. It's "ca-gregorian.json" for "fa" locale

Comment (San4es) [reply testing/helpers/l10n/cldrCalendarDataFa.js]: @ksercs @nightskylark 
Wouldn't it be better to add this package to "devDependencies" and load these files from node_modules? Like the "cldr-numbers-full" package: https://github.com/DevExpress/DevExtreme/blob/c0f577ef779a3f320cc9501c3e2bb154d2a600f5/package.json#L53

Comment (nightskylark) [reply testing/helpers/l10n/cldrCalendarDataFa.js]: https://unpkg.com/browse/devextreme-cldr-data@1.0.2/fa.js

**Review by San4es — CHANGES_REQUESTED** [review]
Let's use the `devextreme-cldr-data` package.

File: testing/tests/DevExpress.localization/localization.intl.widgets.tests.js [hunk 2]
```diff
@@ -23,24 +26,110 @@ const commonEnvironment = {
...
+
+                assert.equal(date, '١١/١١/٢٠١٥', 'date is localized');
+            } catch(e) {
+                assert.ok(false, 'Error occured: ' + e.message);
+            } finally {
+                locale(currentLocale);
+            }
+        });
+
+        QUnit.test('DateBox should not raise error when digits are Farsi digits', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.localization/localization.intl.widgets.tests.js]:
```suggestion
        QUnit.test('DateBox should not raise error when digits are Farsi digits (T867867)', function(assert) {
```

File: testing/tests/DevExpress.localization/localization.intl.widgets.tests.js [hunk 3]
```diff
@@ -23,24 +26,110 @@ const commonEnvironment = {
 
 QUnit.module('Intl localization', () => {
     QUnit.module('DateBox', commonEnvironment, () => {
-        QUnit.test('DateBox should not raise error when digits are not default arabic digits', function(assert) {
+        QUnit.test('DateBox should not raise error when digits are true arabic digits', function(assert) {
```
Comment (San4es) [inline testing/tests/DevExpress.localization/localization.intl.widgets.tests.js]:
```suggestion
        QUnit.test('DateBox should not raise error when digits are true arabic digits (T851630)', function(assert) {
```

File: testing/tests/DevExpress.localization/localization.intl.widgets.tests.js [hunk 4]
```diff
@@ -23,24 +26,110 @@ const commonEnvironment = {
...
+
+        QUnit.test('DateBox should not raise error when digits are Farsi digits', function(assert) {
+            const currentLocale = locale();
+            try {
+                locale('fa-u-ca-gregory');
+                const $dateBox = $('#dateBox').dxDateBox({
+                    value: new Date(2015, 10, 10),
+                    type: 'date',
+                    pickerType: 'calendar',
+                    useMaskBehavior: true,
```
Comment (San4es) [inline testing/tests/DevExpress.localization/localization.intl.widgets.tests.js]:
```suggestion
                    useMaskBehavior: true
```

File: testing/tests/DevExpress.localization/validation.tests.js [hunk 5]
```diff
@@ -3,13 +3,19 @@ require('localization/globalize/number');
 require('localization/globalize/currency');
 require('localization/globalize/date');
 require('localization/globalize/message');
-require('../../helpers/l10n/cldrNumberDataRu.js');
+const cldrData = [
+    require('../../../node_modules/devextreme-cldr-data/ru.json!json')
```
Comment (San4es) [inline testing/tests/DevExpress.localization/validation.tests.js:7]: Can we use [ru.js](https://unpkg.com/browse/devextreme-cldr-data@1.0.2/ru.js) instead? This script loads CLDR data.

File: testing/tests/DevExpress.localization/localization.globalize.tests.js [hunk 6]
```diff
@@ -89,191 +109,192 @@ define(function(require, exports, module) {
...
+                'Array of month names');
+        });
 
-    QUnit.test('getDayNames', function(assert) {
-        assert.deepEqual(dateLocalization.getDayNames(),
-            ['воскресенье', 'понедельник', 'вторник', 'среда', 'четверг', 'пятница', 'суббота'],
-            'Array of day names');
-    });
+        QUnit.test('getPeriodNames', function(assert) {
+            assert.deepEqual(dateLocalization.getPeriodNames(), ['AM', 'PM'], 'Array of period names');
```
Comment (San4es) [inline testing/tests/DevExpress.localization/localization.globalize.tests.js:140]: This change for the "ru" locale looks strange to me.

File: testing/tests/DevExpress.localization/localization.globalize.tests.js [hunk 7]
```diff
@@ -22,11 +22,11 @@ SystemJS.config({
 });
 
 define(function(require, exports, module) {
-    require('../../helpers/l10n/cldrNumberDataRu.js');
-    require('../../helpers/l10n/cldrNumberDataDe.js');
-    require('../../helpers/l10n/cldrCalendarDataRu.js');
-    require('../../helpers/l10n/cldrCurrencyDataRu.js');
-    require('../../helpers/l10n/cldrCurrencyDataEn.js');
+    const cldrData = [
```
Comment (San4es) [inline testing/tests/DevExpress.localization/localization.globalize.tests.js:25]: Can we use [ru.js](https://unpkg.com/browse/devextreme-cldr-data@1.0.2/ru.js) instead?

File: testing/tests/DevExpress.localization/localization.globalize.widgets.tests.js [hunk 8]
```diff
@@ -2,13 +2,19 @@ const likelySubtags = require('../../../node_modules/cldr-core/supplemental/like
 const numberingSystems = require('../../../node_modules/cldr-core/supplemental/numberingSystems.json!');
 const Globalize = require('globalize');
 
+const cldrData = [
+    require('../../../node_modules/devextreme-cldr-data/fa.json!json'),
```
Comment (San4es) [inline testing/tests/DevExpress.localization/localization.globalize.widgets.tests.js:6]: Can we use js files instead?

File: js/localization/globalize/date.js [hunk 9]
```diff
@@ -709,11 +714,11 @@ if(Globalize && Globalize.formatDate) {
             let formatCacheKey;
 
             if(typeof (format) === 'function') {
-                return format(date);
+                return this.removeRtlMarks(format(date));
```
Comment (nightskylark) [inline js/localization/globalize/date.js]: In this case format is a custom function. I don't think we should modify it's result.

File: testing/tests/DevExpress.localization/localization.intl.widgets.tests.js (hunk 2 above)
Comment (ksercs) [reply testing/tests/DevExpress.localization/localization.intl.widgets.tests.js]: fixed

File: testing/tests/DevExpress.localization/localization.intl.widgets.tests.js (hunk 3 above)
Comment (ksercs) [reply testing/tests/DevExpress.localization/localization.intl.widgets.tests.js]: fixed

File: testing/tests/DevExpress.localization/validation.tests.js (hunk 5 above)
Comment (nightskylark) [reply testing/tests/DevExpress.localization/validation.tests.js:7]: No, we can't. It expects `Globalize` to be in global (for bundles approach only).

File: testing/tests/DevExpress.localization/localization.intl.widgets.tests.js (hunk 4 above)
Comment (ksercs) [reply testing/tests/DevExpress.localization/localization.intl.widgets.tests.js]: fixed

File: testing/tests/DevExpress.localization/localization.globalize.tests.js (hunk 6 above)
Comment (ksercs) [reply testing/tests/DevExpress.localization/localization.globalize.tests.js:140]: Last 'devextreme-cldr-data' package version contains 'AM, PM' as period names for some languages, including ru, de, etc. This test has been written to check that the localization data is taken from the appropriate package. So the test has not lost its meaning.

File: testing/tests/DevExpress.localization/validation.tests.js (hunk 5 above)
Comment (San4es) [reply testing/tests/DevExpress.localization/validation.tests.js:7]: Ok

File: js/localization/globalize/date.js (hunk 9 above)
Comment (ksercs) [reply js/localization/globalize/date.js]: got it, thanks

**Review by San4es — DISMISSED** [review]
LGTM

**Review by nightskylark — APPROVED** [review]
LGTM

**Review by DokaRus — APPROVED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**